src/System.Private.CoreLib/shared/System/MemoryExtensions.cs

```diff
@@ -537,11 +537,32 @@ public static int IndexOfAny<T>(this Span<T> span, ReadOnlySpan<T> values)
         {
             if (typeof(T) == typeof(byte))
             {
-                return SpanHelpers.IndexOfAny(
-                    ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(span)),
-                    span.Length,
-                    ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(values)),
-                    values.Length);
+                ref byte valueRef = ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(values));
+                if (values.Length == 2)
```

Comment:
Should we add a `values.Length == 1` check and call `IndexOf` or is it not worth penalizing the common cases?

Reply:
Had it at one point https://github.com/dotnet/coreclr/pull/20738#discussion_r229924110

Its still goes a Vectorized route and is still single pass (as there is only one item); it has a little bit extra overhead from an unnecessary loop of one iteration; so its only about fixed costs rather than a different technique being used. 

The caller has already added overhead by creating a ReadOnlySpan or array of a single element to call this overhead; rather than just call `.IndexOf`, am not sure in what circumstances you'd do this?